### DISCUSSION TITLE
Adaptation for Smaller ID Syringes

### DISCUSSION THREAD

**Discussion by bhclowers** [Karpova-Lab/syringe-pump#5] [Q&A] [answered]
This is a fantastic contribution.  I have not interrogated the source code in detail but I was wondering if you could comment briefly on whether this syringe pump would be compatible (with the appropriate modifications to the syringe holder) with smaller ID syringes and delivery?  I'm interested in delivering a low uL/min of solution for an expeirments using a syring similar to the SGE models. 

https://www.fishersci.com/shop/products/sge-gas-tight-syringe-removable-needle-fixed-needle-250-bevel-point-style/sg006279

**Comment by alustig3** [comment, marked as answer]
Thanks!

Yes the software is compatible with other syringe sizes. You would need to add a resolution definition for your syringe at the top of the file, and then set the resolution on line 48.

 https://github.com/Karpova-Lab/syringe-pump/blob/7ec31eefbf4c6f440344e53de621482b5488047b/Software/syringeTFT/syringeTFT.ino#L12

https://github.com/Karpova-Lab/syringe-pump/blob/7ec31eefbf4c6f440344e53de621482b5488047b/Software/syringeTFT/syringeTFT.ino#L48

[The syringe you mentioned](https://www.labicom.cz/cogwpspogd/uploads/2016/07/SGE-syringes-new.pdf) above has a 2.3mm inner diameter. You can use [this calculator](https://repl.it/@alustig3/SyringeResCalc)  to get the new resolution for that syringe or others by modifying the `innerDiameter` variable. So your software changes would look something like:
``` c++
#define TWO_FIFTY_MICRO 0.0104
const float resolution = TWO_FIFTY_MICRO;
```
### Hardware modifications
Hardware modifications may include:
- reducing size of cradle diameter
- changing shapes of the clamps
- moving the position of the retraction limit switch (since the travel distance of the plunger will be different)

Here is an old 10mL syringe design that may be useful to you?
[10 mL.zip](https://github.com/Karpova-Lab/syringe-pump/files/5307065/10.mL.zip)

It is similar to the picture below but with a smaller syringe. It uses binder clips to hold the syringe in place.
![](https://www.andybuilds.com/work/syringe-pump/pics/v2_side.jpg)
